src/vaults/AsyncRequestManager.sol

```diff
@@ -202,22 +202,34 @@ contract AsyncRequestManager is BaseRequestManager, IAsyncRequestManager {
         ShareClassId scId,
         address user,
         AssetId assetId,
-        uint128 assets,
-        uint128 shares
+        uint128 fulfilledAssets,
+        uint128 fulfilledShares,
+        uint128 cancelledAssets
     ) public auth {
         IAsyncVault vault_ = IAsyncVault(address(vault[poolId][scId][assetId]));
-
         AsyncInvestmentState storage state = investments[vault_][user];
+
         require(state.pendingDepositRequest != 0, NoPendingRequest());
+        if (cancelledAssets > 0) {
+            require(state.pendingCancelDepositRequest == true, NoPendingRequest());
```

Comment:
Not sure if we want to keep this check at all. In practice, this should never throw. Curious about your opinion.

Reply:
No reason to not have it imo

Reply:
It could throw in case of message reordering.

Reply:
> It could throw in case of message reordering.

I don't see how that's possible for this particular case because `state.pendingCancelDepositRequest == false` means there cannot be any pending cancellation request for the user anywhere in the system. Once this value is set to `true`, there will be exactly one fulfilled message with non-zero cancelled amount.

Independent of that, I agree there is now downside in keeping it.